content/articles/online-machine-learning-with-river-python/index.md

```diff
@@ -0,0 +1,446 @@
+
+We have imported the following packages:
+
+#### MultinomialNB
+
+This is a Naive Bayes method that is used for text classification. This contains the algorithm that is useful in the building of our model.
+
+For a detailed understanding about MultinomialNB [click here](https://scikit-learn.org/stable/modules/generated/sklearn.naive_bayes.MultinomialNB.htmll)
+
+#### BagOfWords
+
+Is used to extract various features in our dataset. Features are the independent variables that are used as input for our model.
+
+It also converts our text inputs into vectors that are more machine-readable. BagOfWords work the same as `CountVectorizer` in offline machine learning.
+
+For a detailed understanding about TFIDF [click here](https://en.wikipedia.org/wiki/Bag-of-words_model)
+
+#### TFIDF
+
+It stands for term frequency-inverse document frequency, shows how words are relevant in a given document, and gives the importance of words.
+
+This is done by measuring the frequency of how words appear in a given document, if a word appears frequently in a document, while not appearing frequently in others, it means this word is more relevant and has a more classification power as compared to words that appear in all the documents.
+
+For a detailed understanding of TFIDF [click here](https://towardsdatascience.com/text-vectorization-term-frequency-inverse-document-frequency-tfidf-5a3f9604da6d)
+
+### Getting the relevant methods and attributes
+
+To use the methods stated earlier we need to add them to our program. To do this we used the following Python function.
+
+```python
+def add_all_attributes(package):
+    subpackages = []
+    submodules = []
+    for i in dir(package):
+        if str(i) not in ["__all__", "__builtins__", "__cached__", "__doc__", "__file__", "__loader__", "__name__", "__package__", "__path__", "__pdoc__", "__spec__", "__version__"]:
+            subpackages.append(i)
+            res = [j for j in dir(eval("river.{}".format(i)))]
+            submodules.append(res)
+    df = pd.DataFrame(submodules)
+    df = df.T
+    df.columns = subpackages
+    res_df = df.dropna()
+    return res_df
+```
+
+We name this function `add_all_attributes` to get all the attributes and methods needed to build our model.
+
+The functios loops through the package and remove the followig: `__all__`, `__builtins__`, `__cached__`, `__doc__`, `__file__`, `__loader__`, `__name__`, `__package__`, `__path__`, `__pdoc__`, `__spec__` and `__version__`.
+
+We remove them since are not used to build our model.
+
+The remaining sub-packages and sub-modules in the list are added using the `subpackages.append()` method.
+
+We then add these sub-packages and submodules into our data frame using the `pd.DataFrame(submodules)` method. The data frame is what will be used to train our model. The data frame will now have all the sub-packages and submodules.
+
+To now get these methods and attributes use this command.
+
+```python
+river_df = add_all_attributes(river)
+```
+
```

Comment:
Have we gotten these methods and attributes? Since this is a tutorial, hold the hand of the reader by using screenshots where necessary. And try and explain the results.